doc/admin/installation.md

```diff
@@ -112,6 +114,11 @@ By default, DAOS and its dependencies are installed under ${daospath}/install.
 The installation path can be modified by adding the PREFIX= option to the above
 command line (e.g., PREFIX=/usr/local).
 
+!!! note
+   If you wish to compile DAOS with clang or use the Intel compilers rather than
+   gcc, set COMPILER=clang or COMPILER=icc on the scons command line.
+   This option is also saved for future compilations.
```

Comment:
Basically all KEY=VALUE options are, we should not document it here only, but better to make a separate note for all options.

We should note "scons --help" is available as well, and I'd prefer not to put the --config=force as most people shouldn't need this.